core/jreleaser-model-impl/src/main/java/org/jreleaser/model/internal/release/BitbucketCloudReleaser.java

```diff
@@ -0,0 +1,283 @@
+            return BitbucketCloudReleaser.this.getMilestone().asImmutable();
+        }
+
+        @Override
+        public Issues getIssues() {
+            return BitbucketCloudReleaser.this.getIssues().asImmutable();
+        }
+
+        @Override
+        public boolean isSkipTag() {
+            return BitbucketCloudReleaser.this.isSkipTag();
+        }
+
+        @Override
+        public boolean isSkipRelease() {
+            return BitbucketCloudReleaser.this.isSkipRelease();
+        }
+
+        @Override
+        public boolean isOverwrite() {
+            return BitbucketCloudReleaser.this.isOverwrite();
+        }
+
+        @Override
+        public Update getUpdate() {
+            return BitbucketCloudReleaser.this.getUpdate().asImmutable();
+        }
+
+        @Override
+        public String getApiEndpoint() {
+            return BitbucketCloudReleaser.this.getApiEndpoint();
+        }
+
+        @Override
+        public boolean isArtifacts() {
+            return BitbucketCloudReleaser.this.isArtifacts();
+        }
+
+        @Override
+        public boolean isFiles() {
+            return BitbucketCloudReleaser.this.isFiles();
+        }
+
+        @Override
+        public boolean isChecksums() {
+            return BitbucketCloudReleaser.this.isChecksums();
+        }
+
+        @Override
+        public boolean isCatalogs() {
+            return BitbucketCloudReleaser.this.isCatalogs();
+        }
+
+        @Override
+        public boolean isSignatures() {
+            return BitbucketCloudReleaser.this.isSignatures();
+        }
+
+        @Override
+        public Active getUploadAssets() {
+            return BitbucketCloudReleaser.this.getUploadAssets();
+        }
+
+        @Override
+        public Integer getConnectTimeout() {
+            return BitbucketCloudReleaser.this.getConnectTimeout();
+        }
+
+        @Override
+        public Integer getReadTimeout() {
+            return BitbucketCloudReleaser.this.getReadTimeout();
+        }
+
+        @Override
+        public String getOwner() {
+            return BitbucketCloudReleaser.this.getOwner();
+        }
+
+        @Override
+        public boolean isEnabled() {
+            return BitbucketCloudReleaser.this.isEnabled();
+        }
+
+        @Override
+        public Map<String, Object> asMap(boolean full) {
+            return unmodifiableMap(BitbucketCloudReleaser.this.asMap(full));
+        }
+
+        @Override
+        public CommitAuthor getCommitAuthor() {
+            return BitbucketCloudReleaser.this.getCommitAuthor().asImmutable();
+        }
+
+        @Override
+        public boolean isPrerelease() {
+            return BitbucketCloudReleaser.this.isPrerelease();
+        }
+    };
+
+    public BitbucketCloudReleaser() {
+        super(org.jreleaser.model.api.release.BitbucketCloudReleaser.TYPE, true);
+        // TODO: viqueen - when do I need to set the properties ? can we enforce that somehow through the API ?
```

Comment:
Here the class sets some defaults. The validator later checks if required values have been set. Also, it applies sensible defaults following Convention over Configuration.

Reply:
gotcha !